### DISCUSSION TITLE
I would like to be control Toggl Track tags within Obsidian

### DISCUSSION THREAD

**Discussion by euphonics** [mcndt/obsidian-toggl-integration#20] [Feature Requests]
I would like to tag my time entries using existing Toggl tags and have the ability to create new tags when needed.

Currently, it seems that project selection and time entry description are supported, but tags are completely inaccessible via the plug-in.

This could be achieved by adding a "tags" field input beneath the "description" text input when you are starting a new timer.

Great plug-in, keeps me out of my web browser and in Obsidian where I need to be. Thank you.

**Comment by mcndt** [comment]
Thanks for your feature request! Since I personally don't use tags, I'd like to discuss the implementation of this feature with you in some more detail.

The Toggl API should make it quite easy to work with tags. The reason I hadn't yet implemented it is that not many people use them, and I wasn't sure how to best show it in the UI.

Is a simple entry field underneath the timer description where you put your tags separated by commas satisfactory? Or would you expect something more fancy (e.g. tag autocompletion).

Would you prefer that tags on your active timer are visible in the sidebar view, or do you prefer them hidden under e.g. a popover?

Feel free to message me on Discord if you prefer to discus the details more directly (username Maximio#6460).

**Comment by euphonics** [comment]
I sent you a friend/dm request on Discord if you want to talk further. @arvn

> Is a simple entry field underneath the timer description where you put
your tags separated by commas satisfactory?

Better than nothing

> Or would you expect something more fancy (e.g. tag autocompletion).

Autocomplete is handy because I don't remember the exact names of my tags.
If I spell a tag wrong in Obsidian I don't wan't to accidentally create new
tags. Auto complete will prevent that problem.

> Would you prefer that tags on your active timer are visible in the
sidebar view, or do you prefer them hidden under e.g. a popover?

I prefer the tags to be visible in the sidebar view, although I suppose
ideally there would be an option to show/hide them in the plug-in settings.

Exposing the current tags would actually be an improvement on the Toggl
track web application, which currently doesn't show me which tags I have
assigned to my current timer.

---

Brief comment on why tags are useful to me and how I use them:
I like to keep track of how much time I spend doing particular activities
for a particular project. I am a (wannabe) screenwriter, and each of my
screenplays has its own project in Toggl track. But a single project is
composed of a bunch of different types of activity.  Research,
brainstorming, outlining, writing pages, editing, etc.

I use tags for these activity categories so that I can get insight not only
into how long I have spent on a particular project, but also into what
types of activity are taking up the most time for a particular project or
across all projects. How long did I spend researching on this project
versus this other project? How much time did I spend editing between the
months of January and March? Etc.

This is much more useful information for reflecting on the process than
simply knowing how much time I spent on the project in general, or relying
upon the unstructured difficult-to-analyze free-text descriptions. I no
longer use Toggl track for work, but this kind of information is also very
useful to have one reporting on the status of a project to a manager or
client.


It's a great plug-in, thanks for your hard work. again, message me on
discord if easier.

**Comment by sirlaughalot90** [comment]
I'd love to see tags implemented. I only use them for one purpose, but a very important one, for my productivity tracking process: I separate my work along the lines of deep and shallow work (cf. Cal Newport). Hence, I tag timers with #deepWork if it requires a lot of concentration, e.g. content creation. Therefore, it would be very useful to be able to see in the sidebar whether I tagged something as deep work because sometimes I forget to set the tag. Auto-complete is not necessary, but would be appreciated.

**Reply by mcndt** [reply]
Thanks for letting me know the importance of tags to your workflow! I’ll add this functionality in the next release.

**Reply by sirlaughalot90** [reply]
Awesome. Thank you!

**Comment by mcndt** [comment]
This feature is now available in release 0.3.0.
